core/thread.c

```diff
@@ -124,6 +124,19 @@ void thread_add_to_list(list_node_t *list, thread_t *thread)
     list->next = new_node;
 }
 
+void thread_remove_from_list(list_node_t *list, thread_t *thread)
```

Comment:
thread_add_to_list() is taking the thread's priority into account and adds the thread so afterwards the list is sorted by priority.

+1 for adding "list_remove()" to list.h! and using that!